odoo/addons/test_read_group/tests/test_fill_temporal.py

```diff
@@ -0,0 +1,746 @@
+                                                                         groupby=['date'])
+        self.assertEqual(groups, expected)
+
+        groups = self.Model.with_context(fill_temporal=True).read_group([],
+                                                                        fields=['date',
+                                                                                'value'],
+                                                                        groupby=['date'])
+        self.assertEqual(groups, expected)
+
+    def test_date_range_and_undefined(self):
+        """Similar test than test_only_with_undefined but with setted dates in more."""
+        self.Model.create({'date': '1916-08-18', 'value': 3})
+        self.Model.create({'date': '1916-08-19', 'value': 4})
+        self.Model.create({'date': '1916-10-18', 'value': 5})
+        self.Model.create({'date': '1916-10-19', 'value': 4})
+        self.Model.create({'date': False, 'value': 13})
+        self.Model.create({'date': False, 'value': 11})
+
+        expected = [{'__domain': ['&',
+                                  ('date', '>=', '1916-08-01'),
+                                  ('date', '<', '1916-09-01')],
+                     'date': 'August 1916',
+                     'date_count': 2,
+                     'value': 7},
+                    {'__domain': ['&',
+                                  ('date', '>=', '1916-09-01'),
+                                  ('date', '<', '1916-10-01')],
+                     'date': 'September 1916',
+                     'date_count': 0,
+                     'value': False},
+                    {'__domain': ['&',
+                                  ('date', '>=', '1916-10-01'),
+                                  ('date', '<', '1916-11-01')],
+                     'date': 'October 1916',
+                     'date_count': 2,
+                     'value': 9},
+                    {'__domain': [('date', '=', False)],
+                     'date': False,
+                     'date_count': 2,
+                     'value': 24}]
+
+        groups = self.Model.with_context(fill_temporal=False).read_group([],
+                                                                         fields=['date',
+                                                                                 'value'],
+                                                                         groupby=['date'])
+        self.assertEqual(groups, [group for group in expected if group['date_count']])
+
+        groups = self.Model.with_context(fill_temporal=True).read_group([],
+                                                                        fields=['date',
+                                                                                'value'],
+                                                                        groupby=['date'])
+
+        self.assertEqual(groups, expected)
+
+    def test_date_range_and_undefined_unordered(self):
```

Comment:
I would merge this test with the former one. The data should be created unordered: it is the "general" case.